src/mono/wasm/runtime/debug.ts

```diff
@@ -149,6 +151,11 @@ export function mono_wasm_debugger_attached(): void {
     if (runtimeHelpers.waitForDebugger == -1)
         runtimeHelpers.waitForDebugger = 1;
     cwraps.mono_wasm_set_is_debugger_attached(true);
+    if (MonoWasmThreads) {
+        const wasmMemory = Module.getMemory();
+        wasmMemory.grow(wasmMemory.buffer.byteLength >>> 16);
```

Comment:
I'm not sure how this is related to debugger or threads. Could you please elaborate ?